### ISSUE TITLE
Child Nodes have the same `Text`

### ISSUE THREAD

**Issue by ash-aldujaili** [lzybkr/ShowPSAst#5]
I tried running the tool on the module itself as follows.
```
Show-Ast .\ShowPSAst.psm1
```
I found the following children nodes with the same texts (why is that? I assume nodes would have distinct IDs), see below; two instances of TypeConstraintAst nodes of exactly the same offsets.
![image](https://user-images.githubusercontent.com/1083829/49098503-05db8700-f23d-11e8-8584-2725799d76a8.png)

Debugging these two nodes, I found the following:

![image](https://user-images.githubusercontent.com/1083829/49098102-25be7b00-f23c-11e8-8485-ccf52c561507.png)

In a nutshell, one $child has 
`TypeNameOfValue System.Management.Automation.Language.AttributeBaseAst` and the other has `TypeNameOfValue : System.Management.Automation.Language.TypeConstraintAst`, yet both map to the same text `TypeConstraintAst [2918,2942)`.

Any thoughts on the above?

**Comment by lzybkr** [comment]
They are more than the same extent, they are the same reference, see [Type property](https://github.com/PowerShell/PowerShell/blob/58bbfe8396889599a5aa991157cf5988753b017a/src/System.Management.Automation/engine/parser/ast.cs#L7442) in `ConvertExpressionAst` and [Attribute property](https://github.com/PowerShell/PowerShell/blob/58bbfe8396889599a5aa991157cf5988753b017a/src/System.Management.Automation/engine/parser/ast.cs#L7328) in the base class `AttributedExpressionAst`.

Maybe it looks a little weird, but I don't see any real problems.

**Comment by ash-aldujaili** [comment]
Thanks for the prompt response. 

I am using the tool to extract structural info from PS scripts from a graph-theoretic point of view. If they are the same reference, would it be safe to ignore the second node ( I am assuming that both point to the same object and its properties)? 

In other words, An AST with two children of identical text (as provided in the tool) is---functionally---equivalent to another AST with one child of the same text?

**Comment by lzybkr** [comment]
Yes, I think it's safe to ignore a second immediate child with identical text.

**Comment by ash-aldujaili** [comment]
Noted, with thanks.
I wonder what the advantage of such a setting is for the parser!

**Comment by lzybkr** [comment]
There is a single field backing both properties, so it's simply a convenience api.

**Comment by lzybkr** [comment]
The derived property also adds a type cast which makes the convenience api even more useful when editing code - giving more appropriate completions.
